deps/openssl.BUILD.bazel

```diff
@@ -123,6 +123,16 @@ configure_make(
     }),
     copts = ["-O2"],
     targets = ["depend", "", "install_sw", "install_ssldirs"],
+    # Fix install names, as openssl's build system might not get them right
+    postfix_script = select({
```

Comment:
Something tells me that openssl's build system does produce the correct libraries. Leaning on the results of linux builds we always get versioned and unversioned `.so` objects (`libcrypto.so` and `libcrypto.so.3`), but I just didn't add them to `out_shared_libs` attribute and relied on `so_symlink` instead. If I am right and openssl does produce versioned dylibs we don't need this. We just need to add versioned libs to the expected outputs and remove so_symlink targets altogether